src/features/todos.ts

```diff
@@ -1,9 +1,30 @@
 import { Todo } from '../types/Todo';
 
-export const actions = {};
+type SetAddAction = {
+  type: 'todos/SET_ADD',
+  payload: Todo[],
+};
+
+const SetAdd = (arr: Todo[]): SetAddAction => ({
+  type: 'todos/SET_ADD',
+  payload: arr,
+});
+
+export const actions = { SetAdd };
+
+const initialTodos: Todo[] = [];
+
+const todosReducer = (
+  state = initialTodos,
+  action: SetAddAction,
+): Todo[] => {
+  switch (action.type) {
+    case 'todos/SET_ADD':
+      return [...action.payload, ...state];
 
-const todosReducer = (): Todo[] => {
-  return [];
+    default:
+      return [...state];
```

Comment:
```suggestion
      return state;
```
no need to make copy here